File: Chapter 03/C#/3.4 OrderProcessor Refactor/Before/OrderProcessor.cs
using System;

namespace Chapter03.OrderProcessorRefactor.Before;

public class OrderProcessor
{
    public string Process(Order order)
    {
        // 1. Validation
        Console.WriteLine("  [Validate] Validating order...");
        if (order.Items.Count == 0 || order.Total <= 0)
        {
            throw new InvalidOperationException("Order is invalid.");
        }

        // 2. Payment Processing
        Console.WriteLine($"  [Payment] Processing payment for ${order.Total}...");
        bool paymentSuccess = true; 

        // 3. Inventory Update & 4. Confirmation Email
        if (paymentSuccess)
        {
            Console.WriteLine("  [Inventory] Updating inventory...");
            Console.WriteLine($"  [Notify] Sending confirmation email to {order.CustomerEmail}...");
            return "Order processed successfully.";
        }
        else
        {
            return "Payment failed.";
        }
    }
}

File: Chapter 03/Python/3.4 OrderProcessor Refactor/Before/order_processor.py
class OrderProcessor:
    def process(self, order):
        # 1. Validation
        print("  [Validate] Validating order...")
        if not order.items or order.total <= 0:
            raise ValueError("Order is invalid.")

        # 2. Payment Processing
        print(f"  [Payment] Processing payment for ${order.total:.2f}...")
        payment_success = True

        # 3. Inventory Update & 4. Confirmation Email
        if payment_success:
            print("  [Inventory] Updating inventory...")
            print(f"  [Notify] Sending confirmation email to {order.customer_email}...")
            return "Order processed successfully."
        else:
            return "Payment failed."